websocket.go

```diff
@@ -56,7 +58,7 @@ func wsHandler(errCh chan error, hub *Hub, w http.ResponseWriter, r *http.Reques
 	hub.clients[conn] = true
 	log.Println(hub.clients)
 
-	go writer(conn)
+	go writer(conn, msgChan)
 
 	go reader(conn, errCh)
```

Comment:
you pass errCh to the reader
then reader push errors into it
then you handle this errors in the gouruting below by deleting client from hub.

all that we really need here is to pass hub into reader and deleting clients inside